src/api/firebase.js

```diff
@@ -71,8 +73,44 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 
 export async function updateItem(listId, docId, itemData) {
 	const docRef = doc(db, listId, docId);
+	const { isChecked, totalPurchases, dateLastPurchased, dateNextPurchased } =
+		itemData;
+	let data;
 
-	await updateDoc(docRef, itemData);
+	const currentDate = new Date();
+	const oldDateLastPurchased = dateLastPurchased
+		? dateLastPurchased.toDate()
+		: currentDate;
+	const oldDateNextPurchased = dateNextPurchased.toDate();
```

Comment:
I put the `old` prefix before the `dateLastPurchased` and `dateNextPurchased` variables so we don't get confused with the new calculations we do for these dates.